### ISSUE TITLE
Issue with flatten_commentforest

### ISSUE THREAD

**Issue by aarasmith** [arshadrr/subreddit-archiver#5]
Hi, I'm running the subreddit archiver on /r/combatfootage and getting an error after successfully archiving 26,299 posts.

command used: subreddit-archiver archive --subreddit combatfootage --file combatfootage.sqlite --credentials credentials.config

least_recent_saved_post_utc: 1559489863.0
subreddit_created_utc: 1347238229.0

traceback: 
Traceback (most recent call last):
  File "C:\Users\XXXX\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\XXXX\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Users\XXXX\AppData\Local\Programs\Python\Python310\Scripts\subreddit-archiver.exe\__main__.py", line 7, in <module>
  File "C:\Users\XXXX\AppData\Local\Programs\Python\Python310\lib\site-packages\subreddit_archiver\main.py", line 61, in main
    archive(args.subreddit, args.file, args.batch_size, args.credentials)
  File "C:\Users\XXXX\AppData\Local\Programs\Python\Python310\lib\site-packages\subreddit_archiver\utils.py", line 4, in clean
    func(*args, **kwargs)
  File "C:\Users\XXXX\AppData\Local\Programs\Python\Python310\lib\site-packages\subreddit_archiver\main.py", line 32, in archive
    get_posts.archive_posts(reddit, db_connection, batch_size)
  File "C:\Users\XXXX\AppData\Local\Programs\Python\Python310\lib\site-packages\subreddit_archiver\get_posts.py", line 115, in archive_posts
    process_post_batch(posts, db_connection)
  File "C:\Users\XXXX\AppData\Local\Programs\Python\Python310\lib\site-packages\subreddit_archiver\get_posts.py", line 81, in process_post_batch
    serializer.flatten_commentforest(post.comments, comments)
  File "C:\Users\XXXX\AppData\Local\Programs\Python\Python310\lib\site-packages\subreddit_archiver\serializer.py", line 163, in flatten_commentforest
    flatten_commentforest(comment.replies, outlist)
  File "C:\Users\XXXX\AppData\Local\Programs\Python\Python310\lib\site-packages\subreddit_archiver\serializer.py", line 163, in flatten_commentforest
    flatten_commentforest(comment.replies, outlist)
  File "C:\Users\XXXX\AppData\Local\Programs\Python\Python310\lib\site-packages\subreddit_archiver\serializer.py", line 163, in flatten_commentforest
    flatten_commentforest(comment.replies, outlist)
  [Previous line repeated 8 more times]
AttributeError: 'MoreComments' object has no attribute 'replies'
Subreddit created on Sun Sep  9 20:50:29 2012
Preparing to archive or continue archival

Thanks!

**Comment by arshadrr** [comment]
Hi, thanks for taking the time to report this, and for including all the relevant details. I will take a look at it in the weekend as I've not the time right now.

For the time being, could you try to reduce the `least_recent_saved_post_utc` by a small amount (perhaps using sqlite3 cli shell)? The utility takes a look at this value then fetches posts older than it, when you're archiving. Since you've managed to archive thus far, I'm guessing it's a specific post that is causing the issue. If it is, reducing the value will allow you to skip this problematic post.

**Comment by aarasmith** [comment]
Thanks for the response! I played around with changing the `least_recent_saved_post_utc` metadata with no success. I tried changing from about a few hours to about a month. Changing this value multiple times started giving me some different very verbose errors with a bunch of html, so I tried changing it on a fresh copy each time. No dice.

I then changed the `<BATCH-SIZE>` argument to 1 and it archived 165 posts before crashing again. This seems strange as the default batch size is 100. I then edited the `least_recent_saved_post_utc` value by about 8 hours and it is now working properly again. I've saved the tracebacks so if you're interested I can pass them along. 

Cheers!

**Comment by arshadrr** [comment]
Hi, I took a look and figured out the cause of the original issue (a silly error on my part) and the issue which prints out some html (pushshift.io was down, and the html was cloudflare informing us of this). It should be fixed now.

If you still have the stack traces, I would appreciate it if you could share them, maybe through something like pastebin.com

Thanks again for taking the time to report this :)

**Comment by aarasmith** [comment]
[Here](https://pastebin.com/qRB9sjLr) are 7 or so tracebacks I saved. Most of the time it seems like just a pushshift issue and when I re-run the same command it works fine. If that doesn't work, a PC restart fixes it sometimes. If that doesn't work I change `batch-size` to 1 and wait until it breaks and then run the command again with default `batch-size` and it works fine from there on. Only twice have I had to actually edit the `least_recent_saved_post_utc` metadata to get it to continue.

Anyway, thanks for the fix. This is a very helpful tool!
